src/workspaces/features/CreateWorkspaceDialog/CreateWorkspaceDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import Button from "core/components/Button";
+import Dialog from "core/components/Dialog";
+import Field from "core/components/forms/Field";
+import Spinner from "core/components/Spinner";
+import CountryPicker from "core/features/CountryPicker";
+import { CountryPicker_CountryFragment } from "core/features/CountryPicker/CountryPicker.generated";
+import useForm from "core/hooks/useForm";
+import { useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { useCreateWorkspaceMutation } from "workspaces/graphql/mutations.generated";
+import { ensureArray } from "core/helpers/array";
+import { useRouter } from "next/router";
+import { CreateWorkspaceError } from "graphql-types";
+
+type CreateWorkspaceDialogProps = {
+  onClose(workspaceId: string | null): void;
```

Comment:
I don't see where you pass a `workspaceId`. In your `onSubmit` function you just redirects the user to the new workspace

Reply:
It was in the case the user click on cancel instead of submit